crates/runtime/src/externref.rs

```diff
@@ -315,19 +325,14 @@ impl VMExternRef {
             let value_ptr = alloc_ptr.cast::<T>();
             ptr::write(value_ptr.as_ptr(), make_value());
 
-            let value_ref: &T = value_ptr.as_ref();
-            let value_ref: &dyn Any = value_ref as _;
-            let value_ptr: *const dyn Any = value_ref as _;
-            let value_ptr: *mut dyn Any = value_ptr as _;
-            let value_ptr = NonNull::new_unchecked(value_ptr);
-
             let extern_data_ptr =
                 alloc_ptr.cast::<u8>().as_ptr().add(footer_offset) as *mut VMExternData;
             ptr::write(
                 extern_data_ptr,
                 VMExternData {
-                    ref_count: UnsafeCell::new(1),
-                    value_ptr,
+                    ref_count: AtomicUsize::new(1),
+                    // cast from `*mut T` to `*mut dyn Any` here
```

Comment:
```suggestion
                    // Cast from `*mut T` to `*mut dyn Any` here.
```

That's a lot easier than the way I was doing the casting :-p

I swear casting to `dyn` is the thing I have the most trouble with in Rust at this point.